pulp_ansible/app/viewsets.py

```diff
@@ -88,6 +89,59 @@ class RoleViewSet(ContentViewSet):
     queryset = Role.objects.all()
     serializer_class = RoleSerializer
     filterset_class = RoleFilter
+    queryset_filtering_required_permission = "ansible.view_role"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_repository_model_or_obj_perms:ansible.view_role",
+            },
```

Comment:
I would follow the same advice I gave for CollectionVersions here as well. Also you should remove the `LOCKED_ROLES`. This is not needed and I think it is breaking your tests.